src/local-history/local-history-manager.ts

```diff
@@ -435,18 +468,41 @@ export class LocalHistoryManager {
      */
     private removeOldestRevision(document: vscode.TextDocument): void {
         if (this.checkRevisionLimit(document)) {
-            const hashedPath = this.getHashedFolderPath(document.uri.fsPath);
+            const hashedPath = this.getHashedRevisionFolderPath(document.uri.fsPath);
             const oldestRevision = this.getOldestRevision(hashedPath);
             if (oldestRevision) {
                 fs.unlinkSync(oldestRevision);
             }
         }
     }
 
-    private checksum(str: string, algorithm: string = 'md5') {
+    public checksum(str: string, algorithm: string = 'md5') {
         return crypto
             .createHash(algorithm)
             .update(str, 'utf8')
             .digest('hex');
     }
+
+    /**
+     * Tracks the deleted files.
+     * @param files array of all the deleted file(s) uri.
+     */
+    private async storeDeletedFiles(files: ReadonlyArray<vscode.Uri>) {
+
+        for (const file of files) {
+            const hashedFolderPath = this.getHashedDeletionFolderPath(file.fsPath);
+            if (!fs.existsSync(hashedFolderPath)) {
+                fs.mkdirSync(hashedFolderPath, { recursive: true });
+            }
+            const filename = path.parse(file.fsPath).base;
+            const filePath = path.join(hashedFolderPath, filename);
+            try {
+                const fileContent = await this.readFile(file.fsPath);
+                await this.writeFile(filePath, fileContent, false);
```

Comment:
@vince-fugnitto @kaiyue0329 we should also make sure the deleted file is read-only right?